static/app/components/modals/addDashboardIssueWidgetModal.tsx

```diff
@@ -0,0 +1,265 @@
+        title: '',
+        interval: '5m',
+        queries: [{...newQuery}],
+        errors: undefined,
+        loading: false,
+      };
+      return;
+    }
+
+    this.state = {
+      title: widget.title,
+      interval: widget.interval,
+      queries: widget.queries,
+      errors: undefined,
+      loading: false,
+    };
+  }
+
+  handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const {
+      api,
+      organization,
+      onAddWidget,
+      onUpdateWidget,
+      widget: previousWidget,
+    } = this.props;
+    this.setState({loading: true});
+    let errors: FlatValidationError = {};
+    const widgetData: Widget = {
+      title: this.state.title,
+      interval: this.state.interval,
+      queries: this.state.queries,
+      displayType: DisplayType.TABLE,
+      type: WidgetType.ISSUE,
+    };
+    try {
+      await validateWidget(api, organization.slug, widgetData);
+      if (typeof onUpdateWidget === 'function' && !!previousWidget) {
+        onUpdateWidget({
+          id: previousWidget?.id,
+          ...widgetData,
+        });
+        addSuccessMessage(t('Updated widget.'));
+      } else if (onAddWidget) {
+        onAddWidget(widgetData);
+        addSuccessMessage(t('Added widget.'));
+      }
+    } catch (err) {
+      errors = mapErrors(err?.responseJSON ?? {}, {});
+      this.setState({errors});
+    } finally {
+      this.setState({loading: false});
+    }
+  };
+
+  handleFieldChange = (field: string) => (value: string) => {
+    const {organization} = this.props;
+    this.setState(prevState => {
+      const newState = cloneDeep(prevState);
+      set(newState, field, value);
+
+      trackAdvancedAnalyticsEvent('dashboards_views.add_widget_modal.change', {
+        from: 'dashboards',
+        field,
+        value,
+        organization,
+      });
+
+      return {...newState, errors: undefined};
+    });
+  };
+
+  handleQueryChange = (widgetQuery: WidgetQuery) => {
+    this.setState(prevState => {
+      const newState = cloneDeep(prevState);
+      set(newState, `queries`, [widgetQuery]);
+
+      return {...newState, errors: undefined};
+    });
+  };
+
+  render() {
+    const {
+      Footer,
+      Body,
+      Header,
+      api,
+      organization,
+      selection,
+      onUpdateWidget,
+      widget: previousWidget,
+      start,
+      end,
+      statsPeriod,
+    } = this.props;
+    const state = this.state;
+    const errors = state.errors;
+
+    // Construct GlobalSelection object using statsPeriod/start/end props so we can render widget graph using saved timeframe from Saved/Prebuilt Query
+    const querySelection: GlobalSelection = statsPeriod
+      ? {...selection, datetime: {start: null, end: null, period: statsPeriod, utc: null}}
+      : start && end
+      ? {...selection, datetime: {start, end, period: '', utc: null}}
+      : selection;
+
+    const isUpdatingWidget = typeof onUpdateWidget === 'function' && !!previousWidget;
```

Comment:
nit: this may read a bit clearer
```suggestion
    const isUpdatingWidget = defined(onUpdateWidget) && !!previousWidget;
```

Reply:
👍  updated!